tests/integration/tokenomics/relay_mining_integration_test.go

```diff
@@ -1,143 +1,214 @@
+	// for the test service.
+	_, err = keepers.ServiceKeeper.UpdateRelayMiningDifficulty(sdkCtx, map[string]uint64{service.Id: 1})
+	require.NoError(t, err)
+
+	// Set the previous relays and rewards to be used to calculate the increase ratio.
+	previousNumRelays := uint64(0)
+	previousRewards := sdkmath.NewInt(0)
+
+	// Set the initial difficulty multiplier to later check that it has increased.
+	difficultyMultiplier := big.NewRat(1, 1)
+
+	// Monotonically increase the number of relays from a very small number
+	// to a very large number.
+	for numRelays := initialNumRelays; numRelays <= maxNumRelays; numRelays *= 2 {
+		getSessionReq := sessiontypes.QueryGetSessionRequest{
+			ApplicationAddress: appAddress,
+			ServiceId:          service.Id,
+			BlockHeight:        sdkCtx.BlockHeight(),
+		}
+		sessionRes, err := keepers.SessionKeeper.GetSession(sdkCtx, &getSessionReq)
+		require.NoError(t, err)
+
+		session := sessionRes.Session
 
-func (s *RelayMiningIntegrationTestSuite) TestComputeNewDifficultyHash_RewardsReflectWorkCompleted() {
-	// Set the shared module param compute_units_to_tokens_multiplier.
-	// TODO_BETA(#826): wait for integration app & suites refactor to be merged.
-	// _, err := s.RunUpdateParam(s.T(),
-	// 	sharedtypes.ModuleName,
-	// 	sharedtypes.ParamComputeUnitsToTokensMultiplier,
-	// 	computeUnitsToTokensMultiplier,
-	// )
-	// require.NoError(s.T(), err)
-
-	// Set the proof params so we never need a proof (for simplicity of this test)
-	// TODO_BETA(#826): wait for integration app & suites refactor to be merged.
-	// _, err = s.RunUpdateParam(s.T(),
-	// 	prooftypes.ModuleName,
-	// 	prooftypes.ParamProofRequestProbability,
-	// 	float32(0),
-	// )
-	// require.NoError(s.T(), err)
-
-	// Set the proof requirement threshold to be VERY high.
-	// TODO_BETA(#826): wait for integration app & suites refactor to be merged.
-	// _, err = s.RunUpdateParam(s.T(),
-	// 	prooftypes.ModuleName,
-	// 	prooftypes.ParamProofRequirementThreshold,
-	// 	&proofRequirementThreshold,
-	// )
-	// require.NoError(s.T(), err)
-
-	// TODO(@red-0ne, #781): Implement this test after the business logic is done.
-
-	/*
 		// Determine the height at which the claim will expire.
-		sharedParams := sharedtypes.DefaultParams()
-		claimWindowSizeBlocks := int64(sharedParams.GetClaimWindowOpenOffsetBlocks() + sharedParams.GetClaimWindowCloseOffsetBlocks())
-		proofWindowSizeBlocks := int64(sharedParams.GetProofWindowOpenOffsetBlocks() + sharedParams.GetProofWindowCloseOffsetBlocks())
+		sessionEndToProofWindowCloseBlocks := sharedtypes.GetSessionEndToProofWindowCloseBlocks(&sharedParams)
+		sessionEndHeight := session.GetHeader().GetSessionEndBlockHeight()
+		claimWindowOpenHeight := sessionEndHeight + int64(sessionEndToProofWindowCloseBlocks) + 1
 
-		app := integrationApp.DefaultApplication
-		supplier := integrationApp.DefaultSupplier
-		service := integrationApp.DefaultService
+		ctxAtHeight := sdkCtx.WithBlockHeight(claimWindowOpenHeight)
```

Comment:
It doesn't look like the original context is needed, I think it's more idiomatic/conventional to assign over the existing context:

```suggestion
		ctx = sdkCtx.WithBlockHeight(claimWindowOpenHeight)
```

If you need a reference to `sdkCtx` below:

```suggestion
		sdkCtx = sdkCtx.WithBlockHeight(claimWindowOpenHeight)
		ctx = sdkCtx
```